syntax = "proto3";
option cc_generic_services = true;
option java_generic_services = true;

import "common.proto";

package reward; 

message RewardPoolItem
{
   common.UniqueId id = 1;
   string name = 2;
   int32 amount = 3;
   int32 amountMax = 4;
}

message RewardPoolItems
{
   repeated RewardPoolItem items = 1;
}

service RewardService 
{
   rpc GetPools (common.EmptyMessage) returns (RewardPoolItems);
   rpc DeletePool (common.UniqueId) returns (common.OperationResultMessage);
   rpc EditPool (RewardPoolItem) returns (common.OperationResultMessage);
   rpc AddPool (RewardPoolItem) returns (common.UniqueId);

   rpc AddPoints (common.IntMessage) returns (common.OperationResultMessage);
}

